### PR TITLE
Authenticate testing

### PR DESCRIPTION
**Pull request by RodPaDev**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Testing

## Change status

- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

These are unit tests, they are tested when tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

File: rest-api/authentication/models/auth-models.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ function register(user) {
 
 function login(credentials) {
     return db('users')
+    .select('username', 'id', 'password')
```
Comment (CarnunMP) [inline rest-api/authentication/models/auth-models.js]: Should probably add a tab here. :))

File: rest-api/authentication/routes/auth-routes.spec.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+const request = require('supertest');
+const server = require('../../../api/server');
+
+function random(option) {
```
Comment (CarnunMP) [inline rest-api/authentication/routes/auth-routes.spec.js]: Nice!

File: rest-api/authentication/routes/auth-routes.spec.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    password: 'password'
+}
+
+
+// Add Register
+
+describe('POST /api/auth/register', () => {
+    it('Create a new User and return ', () => {
+        return request(server)
+            .post('/api/auth/register')
```
Comment (CarnunMP) [inline rest-api/authentication/routes/auth-routes.spec.js]: Hmm... reason for preference over async/await, here? :)

Comment (CarnunMP) [reply rest-api/authentication/routes/auth-routes.spec.js]: *and elsewhere?

Comment (RodPaDev) [reply rest-api/authentication/routes/auth-routes.spec.js]: I think it looks a lot cleaner in this case

**Pull request merged**